crates/wit-parser/src/lib.rs

```diff
@@ -314,24 +346,27 @@ pub struct Interface {
     ///
     /// Export names are listed within the types themselves. Note that the
     /// export name here matches the name listed in the `TypeDef`.
+    #[serde(serialize_with = "serialize_id_map")]
     pub types: IndexMap<String, TypeId>,
 
     /// Exported functions from this interface.
     pub functions: IndexMap<String, Function>,
 
     /// The package that owns this interface.
+    #[serde(serialize_with = "serialize_optional_id")]
     pub package: Option<PackageId>,
 }
 
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Serialize)]
 pub struct TypeDef {
     pub docs: Docs,
     pub kind: TypeDefKind,
     pub name: Option<String>,
```

Comment:
Could this be `serde(skip)`'d if the name is `None`?

Reply:
This is done in https://github.com/ydnar/wasm-tools/commit/68b52f6ed5cbb7f9bdbe8830426bbabaa0fd6127

Reply:
@alexcrichton would reordering some of the fields in the `Resolve` or child structs be a problem?

I’d like to reorganize a little, moving `name` above `docs`, for instance, so the serialization is a bit more human-readable.